### PR TITLE
MINIFICPP-1794 Remove sensitive properties from agent manifest

### PR DESCRIPTION
**Pull request by lordgamez**
https://issues.apache.org/jira/browse/MINIFICPP-1794

----------------------------------------------
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: libminifi/src/core/state/nodes/SupportedOperations.cpp [hunk 1]
```diff
@@ -62,11 +63,27 @@ void SupportedOperations::addProperty(SerializedResponseNode& properties, const
...
+  sensitive_properties.assign(Configuration::DEFAULT_SENSITIVE_PROPERTIES.begin(), Configuration::DEFAULT_SENSITIVE_PROPERTIES.end());
+  if (configuration_reader_) {
+    const auto additional_sensitive_props_list = configuration_reader_(Configuration::nifi_sensitive_props_additional_keys);
+    if (additional_sensitive_props_list) {
+      std::vector<std::string> additional_sensitive_properties = utils::StringUtils::split(*additional_sensitive_props_list, ",");
+      return Configuration::mergeProperties(sensitive_properties, additional_sensitive_properties);
+    }
+  }
+  return sensitive_properties;
+}
```
Comment (adamdebreceni) [inline libminifi/src/core/state/nodes/SupportedOperations.cpp:77]: it seems to me, that there are three separate instances of building up the sensitive properties list, could we somehow unify these?

Comment (lordgamez) [reply libminifi/src/core/state/nodes/SupportedOperations.cpp:77]: I was thinking about this, when I checked how these properties are used in the encrypt-config. There was one major difference though that the encrypt-config reads these properties from a ConfigFile object which is a PropertiesFile representation and not from a whereas we only have a function object available to read a single property from a Configuration object in the SupportedProperties.

As for the tests we also have a similar problem where we only pass a function object to the SupportedOperations to reach the configuration properties while in the tests we have the actual Configuration object available, so it's not that easy to generalize it, but I was also worried about to use the same code that we are testing in the test as well.

Comment (adamdebreceni) [reply libminifi/src/core/state/nodes/SupportedOperations.cpp:77]: the least common denominator seems to be the function object, we could go for a static `Configuration::getSensitiveProperties(std::function<std::string(const std::string)> reader)`

Comment (lordgamez) [reply libminifi/src/core/state/nodes/SupportedOperations.cpp]: That's a fair point, extracted the function in 17c2db229f70f3453489cc9e2a7a96a65422da15

File: libminifi/src/core/state/nodes/SupportedOperations.cpp [hunk 2]
```diff
@@ -64,9 +65,12 @@ void SupportedOperations::addProperty(SerializedResponseNode& properties, const
...
   std::vector<std::unordered_map<std::string, std::string>> supported_config_updates;
-  for (const auto& config_property : minifi::Configuration::CONFIGURATION_PROPERTIES) {
-    if (!update_policy_controller_ ||
-        (update_policy_controller_ && update_policy_controller_->canUpdate(std::string(config_property.name)))) {
+  for (const auto& config_property : Configuration::CONFIGURATION_PROPERTIES) {
+    auto sensitive_properties = Configuration::getSensitiveProperties(configuration_reader_);
+    if (ranges::find(sensitive_properties, config_property.name) != ranges::end(sensitive_properties)) {
+      continue;
+    }
+    if (!update_policy_controller_ || (update_policy_controller_ && update_policy_controller_->canUpdate(std::string(config_property.name)))) {
```
Comment (martinzink) [inline libminifi/src/core/state/nodes/SupportedOperations.cpp:73]:
```suggestion
    if (!update_policy_controller_ || update_policy_controller_->canUpdate(std::string(config_property.name))) {
```

The second checking of update_policy_controller_ seems unnecessary.

**Review by martinzink — COMMENTED** [review]
LGTM, just a small suggestion

File: libminifi/src/core/state/nodes/SupportedOperations.cpp (hunk 2 above)
Comment (lordgamez) [reply libminifi/src/core/state/nodes/SupportedOperations.cpp]: Good point, updated in 8561e67e7453596a5158f2ab937fc724ec551974

**Review by martinzink — APPROVED** [review]

**Review by arpadboda — APPROVED** [review]

**Pull request closed without merging**